### PR TITLE
Vendir packages CRDs locally

### LINKED ISSUE #2679: Packages should vendir CRD files locally instead of duplicating
**Describe the feature request**

Some packages duplicate CRD files from `config/crd/bases` in their package manifests (examples [1](https://github.com/vmware-tanzu/tanzu-framework/blob/main/packages/featuregates/bundle/config/upstream/feature-crd.yaml), [2](https://github.com/vmware-tanzu/tanzu-framework/blob/main/packages/featuregates/bundle/config/upstream/featuregate-crd.yaml), [3](https://github.com/vmware-tanzu/tanzu-framework/blob/main/packages/cliplugins/bundle/config/cliplugin-crd.yaml)). These files should be `vendir`ed to ensure any updates to CRDs (generally due to updated `controller-tools`) reflects in the packages immediate and avoids lagging behind. An example of doing that is shown [here](url).

Additionally, this needs a Makefile check to make sure `vendir` is run regularly i.e. run `make package-vendir-sync` and verify there is no diff.

**Describe alternatives you've considered**

Continue to duplicate and risk CRDs being out of sync.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by Vandy-P**
Signed-off-by: Vandana Pathak <pathakv@sjc-ubu-eng-127.vmware.com>

### What this PR does / why we need it
We currently use git to vendir these CRDs, but it can get out of date
if CRDs are updated and git SHA in vendir.yml isn't. This makes sure
CRDs are always current because we reference locally.
### Which issue(s) this PR fixes

Fixes #2679 

### Describe testing done for PR

-  make package-vendir-sync
featuregates-package
└── config
    ├── upstream
    │   ├── certificates.yaml
    │   ├── crds
    │   │   ├── config.tanzu.vmware.com_featuregates.yaml
    │   │   └── config.tanzu.vmware.com_features.yaml
    │   ├── helpers.star
    │   ├── rbac.yaml
    │   ├── service.yaml
    │   ├── tanzu-featuregates-manager.yaml
    │   └── webhook.yaml
    └── values.yaml
        
- After changing base CRD and not changes package CRD, build is failing with following error
   **"Please verify if package CRD changes are intended and commit the diffs if so"**   
   

### Release note

```release-note
None
```

### CODE REVIEW

**Review by codegold79 — APPROVED** [review]
I like the clear error message and instructions on how to remedy the problem.

**Review by rajathagasthya — APPROVED** [review]
Thanks for the changes! I'll let @vijaykatam or @maralavi give a final approval before merge.

File: packages/addons-manager/bundle/config/upstream/crds/cni.tanzu.vmware.com_antreaconfigs.yaml [hunk 1]
```diff
@@ -65,6 +65,22 @@ spec:
                 properties:
                   config:
                     properties:
+                      antreaProxy:
```
Comment (vijaykatam) [inline packages/addons-manager/bundle/config/upstream/crds/cni.tanzu.vmware.com_antreaconfigs.yaml:68]: @knabben - Looks like antrea configs were out of date. Do we need to port back to release-0.25 branch for TKGS?

**Review by vijaykatam — APPROVED** [review]
Change looks good and this fixes the out of sync CRD's issue. On a separate note, this will make refactor effort of tanzu-framework slightly more complicated because the packages are now referring to directories that are going to be split up.  @rajathagasthya @yharish991 - Should we look at moving the packages to their component directories?

**Pull request merged**